lib_xua/src/core/endpoint0/xua_endpoint0.c

```diff
@@ -627,75 +627,95 @@ void XUA_Endpoint0_loop(XUD_Result_t result, USB_SetupPacket_t sp, chanend c_ep0
                                     {
-                                        outuint(c_aud_ctl, g_chanCount_Out_HS[sp.wValue-1]);                 /* Channel count */
-                                        outuint(c_aud_ctl, g_subSlot_Out_HS[sp.wValue-1]);    /* Subslot */
-                                        outuint(c_aud_ctl, g_sampRes_Out_HS[sp.wValue-1]);    /* Resolution */
+                                        outct(c_aud_ctl, SET_STREAM_OUTPUT_START);
+                                        outuint(c_aud_ctl, g_dataFormat_Out[newStreamAlt_Out-1]);        /* Data format (PCM/DSD) */
+
+                                        if(g_curUsbSpeed == XUD_SPEED_HS)
+                                        {
+                                            outuint(c_aud_ctl, g_chanCount_Out_HS[newStreamAlt_Out-1]);  /* Channel count */
+                                            outuint(c_aud_ctl, g_subSlot_Out_HS[newStreamAlt_Out-1]);    /* Subslot */
+                                            outuint(c_aud_ctl, g_sampRes_Out_HS[newStreamAlt_Out-1]);    /* Resolution */
+                                        }
+                                        else
+                                        {
+                                            outuint(c_aud_ctl, NUM_USB_CHAN_OUT_FS);                     /* Channel count */
+                                            outuint(c_aud_ctl, g_subSlot_Out_FS[newStreamAlt_Out-1]);    /* Subslot */
+                                            outuint(c_aud_ctl, g_sampRes_Out_FS[newStreamAlt_Out-1]);    /* Resolution */
+                                        }
                                     }
                                     else
                                     {
-                                        outuint(c_aud_ctl, NUM_USB_CHAN_OUT_FS);              /* Channel count */
-                                        outuint(c_aud_ctl, g_subSlot_Out_FS[sp.wValue-1]);    /* Subslot */
-                                        outuint(c_aud_ctl, g_sampRes_Out_FS[sp.wValue-1]);    /* Resolution */
+                                        outct(c_aud_ctl, SET_STREAM_OUTPUT_STOP);
                                     }
-
                                     /* Handshake */
                                     chkct(c_aud_ctl, XS1_CT_END);
-                                }
-                            }
+
+                                    UserAudioStreamState(g_curStreamAlt_In > 0, g_curStreamAlt_Out > 0);
+                                } /* (g_curStreamAlt_Out != newStreamAlt_Out) */
+                            } /* (newStreamAlt_Out <= OUTPUT_FORMAT_COUNT) */
                             break;
 #endif
 
-#if (NUM_USB_CHAN_IN > 0) && (AUDIO_CLASS == 2)
+#if (NUM_USB_CHAN_IN > 0)
                         case INTERFACE_NUMBER_AUDIO_INPUT:
+                            (void) sp.wIndex; /* Compiler bug - this dummy statement avoids an error when int declared after case */
```

Comment:
do you just need to create some scope ( { } )?

Reply:
Yes, that worked thanks. A very strange bug though. Specifically doesn't like declaration after case.